### PR TITLE
Implemented Delete Selected button functionality

### LINKED ISSUE #1416: Implement the multi delete in load game menu
"Delete Selected" button in the load menu is currently always disabled.
It should have a confirmation popup.

### LINKED ISSUE #1458: Implement delete selected button
SaveManagerGUI.cs "private Button deleteSelectedButton;" is unused

- [ ] should be implemented so that it is enabled when some saves are selected.
- [ ] And the actual delete selected functionality

### LINKED ISSUE #1562: Deleting a save doesn't update the number of selected saves
https://github.com/Revolutionary-Games/Thrive/issues/1458#issuecomment-678631088

### PR DESCRIPTION
**Pull request by DTanxxx**
Delete Selected button functionality implemented, as well as a confirm dialog. Also added a call to UpdateSelectedCount in _Process after getSaveCountTask is completed, in SaveManagerGUI, so that the selected count is shown and updated correctly.

closes #1416
closes #1562
closes #1458

### CODE REVIEW

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Fine, I'll do it myself (and review this).
Looks like everything else is working except when I selected two saves and hit the trashcan button on one of those the selected number of saves text wasn't updated:
Kuvakaappaus - 2020-08-24 18-39-49

**Comment by DTanxxx** [comment]
If the trashcan button is pressed the SaveList class will handle the deletion. Since SaveList does not have a reference to the selectedItemCount label in SaveManagerGUI, it cannot update the label right after deletion and requires the Refresh button to be pressed. The way the selectedItemCount label is updated is very similar to totalSaveCount and totalSaveSize labels, where Refresh button needs to be pressed in order to update. However if updating selectedItemCount straight away is the desired effect then I can make `UpdateSelectedCount` method in SaveManagerGUI class a public static method so that I can reference it in SaveList class in _Process

**Comment by hhyyrylainen** [comment]
If the top level component needs to react to an event from a child component, the child component should define a new signal.
In fact the list item already communicates the deletion through a signal:
https://github.com/Revolutionary-Games/Thrive/blob/7a2889789a89aafbb8ee700046f4271bf58aaae5/src/saving/SaveListItem.cs#L67
Though it probably doesn't reach the save manager, so the list should also define such a signal, like OnItemsChanged so that the save manager can listen for that.

File: src/saving/SaveList.cs [hunk 1]
```diff
@@ -146,6 +149,6 @@ private void OnConfirmDelete()
         SaveHelper.DeleteSave(saveToBeDeleted);
         saveToBeDeleted = null;
 
-        Refresh();
```
Comment (hhyyrylainen) [inline src/saving/SaveList.cs:149]: Removing this refresh breaks refresh in the "Load Game" from the pause menu.

File: src/saving/SaveManagerGUI.cs [hunk 2]
```diff
@@ -102,14 +107,16 @@ public override void _Process(float delta)
             Math.Round((float)info.diskSpace / Constants.MEBIBYTE, 2).ToString(CultureInfo.CurrentCulture) + " MiB";
 
         refreshing = false;
+
+        UpdateSelectedCount();
```
Comment (hhyyrylainen) [inline src/saving/SaveManagerGUI.cs:111]: I would like these two lines to be above the `refreshing = false` line as conceptually I imagine that to be like the end of this method, marking things as done.

File: src/saving/SaveManagerGUI.cs [hunk 3]
```diff
@@ -150,6 +158,24 @@ private void LoadFirstSelectedSave()
...
+        deleteSelectedConfirmDialog.DialogText =
+            "Deleting the selected save(s) cannot be undone, are you sure you want to permanently delete " +
+            $"{Selected.Count} save(s)?";
+        deleteSelectedConfirmDialog.PopupCenteredMinsize();
+    }
+
+    private void OnConfirmDeleteSelected()
+    {
+        GD.Print("Deleting save(s): ", string.Join(", ", Selected.Select(item => item.SaveName).ToList()));
+        Selected.ForEach(item => SaveHelper.DeleteSave(item.SaveName));
```
Comment (hhyyrylainen) [inline src/saving/SaveManagerGUI.cs:172]: I'd add a blank line before this line to give the code more room to breathe.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

**Review by hhyyrylainen — APPROVED** [review]
I didn't find any further problems.

**Pull request merged**